### ISSUE TITLE
Build error building against libreseel-dev

### ISSUE THREAD

**Issue by elbourki** [netdata/netdata#8796]
##### Bug report summary

Getting the following error while trying to install netdata in Alpine, I followed the steps in the official [guide](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/alpine).

```
libnetdata/socket/security.c: In function 'tls_select_version':
libnetdata/socket/security.c:71:16: error: 'TLS1_3_VERSION' undeclared (first use in this function); did you mean 'TLS1_2_VERSION'?
   71 |         return TLS1_3_VERSION;
      |                ^~~~~~~~~~~~~~
      |                TLS1_2_VERSION
libnetdata/socket/security.c:71:16: note: each undeclared identifier is reported only once for each function it appears in
libnetdata/socket/security.c:74:12: error: 'TLS_MAX_VERSION' undeclared (first use in this function); did you mean 'TLS1_2_VERSION'?
   74 |     return TLS_MAX_VERSION;
      |            ^~~~~~~~~~~~~~~
      |            TLS1_2_VERSION
make[2]: *** [Makefile:4470: libnetdata/socket/security.o] Error 1
```

Running `openssl version` returns this:

> OpenSSL 1.1.1g  21 Apr 2020 (Library: OpenSSL 1.1.1d  10 Sep 2019)

**Comment by thiagoftsm** [comment]
Hi @elbourki ,

What is the alpine version you are using?

best regards!

**Comment by elbourki** [comment]
Hey @thiagoftsm!
Thanks for the quick reply, my Alpine version is 3.11.

**Comment by thiagoftsm** [comment]
I compiled now the latest Netdata code on Alpine 3.11 and I could compile without any problem, I have the following packages installed on my Alpine VM:

```
openssl-dev-1.1.1f-r0 x86_64 {openssl} (OpenSSL) [installed]
openssl-dev-1.1.1d-r3 x86_64 {openssl} (OpenSSL)
openssl-1.1.1d-r3 x86_64 {openssl} (OpenSSL) [installed]
openssl-1.1.1f-r0 x86_64 {openssl} (OpenSSL) [upgradable from: openssl-1.1.1d-r3]
```

@prologic and @Ferroin  I remember that I used the instructions that you wrote in our `kernel-collector` repository for I install my VM. Do we need to update our documentation? I am not sure whether the packages from a repository can describe the problems that @elbourki is having, because I do not use Alpine every day.

**Comment by elbourki** [comment]
Hi again, @thiagoftsm.

I think I figured it out, I had `libressl-dev` as a dependency, here's a minimal _Dockerfile_ to reproduce the issue:

```
FROM alpine:latest

RUN apk add alpine-sdk bash curl libuv-dev zlib-dev util-linux-dev libmnl-dev gcc make autoconf automake pkgconfig python logrotate libressl-dev
RUN mkdir /netdata
WORKDIR /netdata
RUN git clone https://github.com/netdata/netdata.git . --depth=100
RUN ./netdata-installer.sh
```

Hopefully this helps identify the issue, even though I'm not sure if this is something you'd want to fix.
Just as a fyi, I was able to install netdata v1.20.0 even with libressl on there :D 

Also, kudos to the team working on this awesome piece of software!

**Comment by prologic** [comment]
Ooooh `libressl-dev`. You should have told us thais :)

I'm not entirely certainl we compile successfully against `libressl`. Let me check...

**Comment by prologic** [comment]
Confirmed. Re-titling this issue appropriately and marking as a bug.

**Comment by prologic** [comment]
Note that I do not believe this is specific to Alpine at all.

**Comment by thiagoftsm** [comment]
Thank you @elbourki for your answer and @prologic for your test.
It is too late in my timezone now, so I won't fix today, but tomorrow morning I will bring a fix for `libressl`.

**Comment by thiagoftsm** [comment]
We merged today a solution for this specific problem.
@elbourki, please, let us know if we fixed the problem.

**Comment by elbourki** [comment]
I'm already using netdata in production (with OpenSSL) but I just built another docker image with libressl and it worked nicely!
Well done, you can close this issue now 👌

**Comment by thiagoftsm** [comment]
Thank you for the feedback @elbourki!
